### PR TITLE
Fix taskmanager - allow tx during sync

### PR DESCRIPTION
**Pull request by shargon**
Related to https://github.com/neo-project/neo/pull/2176/files#r548443610

### CODE REVIEW

File: src/neo/Network/P2P/TaskManager.cs [hunk 1]
```diff
@@ -94,9 +94,6 @@ private void OnNewTasks(InvPayload payload)
         {
             if (!sessions.TryGetValue(Sender, out TaskSession session))
                 return;
-            // Do not accept payload of type InventoryType.TX if not synced on best known HeaderHeight
-            if (payload.Type == InventoryType.TX && Blockchain.Singleton.Height < sessions.Values.Max(p => p.LastBlockIndex))
-                return;
```
Comment (Tommo-L) [inline src/neo/Network/P2P/TaskManager.cs:99]: The reason for this line is that when a node falls below the best block height, it's better to synchronize the blocks first.

**Comment by shargon** [comment]
@erikzhang we thought that it can solve the problem, what do you think?

**Comment by superboyiii** [comment]
We could apply it at this moment.

**Review by superboyiii — APPROVED** [review]

**Comment by erikzhang** [comment]
Don't we add the headers back?

**Comment by Tommo-L** [comment]
I think we don't need to add header back, we can optimize it in other ways. Like https://github.com/neo-project/neo/pull/2176#issuecomment-763318201

**Comment by erikzhang** [comment]
> I think we don't need to add header back, we can optimize it in other ways.

Let's do it directly.

**Pull request closed without merging**